src/ns-migration.sol

```diff
@@ -0,0 +1,190 @@
+// Copyright (C) 2020 Centrifuge
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity >=0.5.15 <0.6.0;
+
+
+// TODO: split interfaces between tests and spell. Exclude all the function that afre only used in tests
+interface TinlakeRootLike {
+    function relyContract(address, address) external;
+}
+
+interface MemberlistLike {
+    function updateMember(address, uint) external;
+}
+
+interface DependLike {
+    function depend(bytes, address) external;
+}
+
+interface FileLike {
+    function file(bytes, address) external;
+}
+
+interface AuthLike {
+    function rely(address) external;
+    function deny(address) external;
+}
+
+interface MigrationLike {
+    function migrate(address) external;
+}
+
+interface ERC20Like {
+    function balanceOf(address) external view returns (uint256);
+    function transferFrom(address, address, uint) external returns (bool);
+    function approve(address, uint) external;
+}
+
+interface ReserveLike {
+    function payout(uint currencyAmount) external;
+}
+
+// spell for: ns2 migration to rev pool with maker support
+// - migrate state & swap contracts: assessor, reserve, coordinator
+// - add & wire mkr adapter contracts: clerk & mgr, spotter, vat
+contract TinlakeSpell {
+
+    bool public done;
+    string constant public description = "Tinlake NS2 migration kovan Spell";
+
+    address constant public ROOT = 0xB86a5F04F5aE402d8547842A29071309C23b47e1;
```

Comment:
Why not only passing the root address from a deployment and fetching all other addresses from the deployer contracts.

I think it would avoid potential wrong contract addresses and no copy paste the next time.

Reply:
unfortunately not always possible, as contract addresses may change